packages/yoroi-extension/app/stores/toplevel/TransactionsStore.js

```diff
@@ -281,8 +283,25 @@ export default class TransactionsStore extends Store<StoresMap, ActionsMap> {
     return result ? result.totalAvailable : 0;
   }
 
+  // This method ensures that at any time, there is only one refreshing process
+  // for each wallet.
+  refreshTransactionData: {|
+    publicDeriver: PublicDeriver<>,
+    localRequest: boolean,
+  |} => Promise<void> = async (request) => {
+    const { publicDeriverId } = request.publicDeriver;
+    if (this.ongoingRefreshing.has(publicDeriverId)) {
+      await this.ongoingRefreshing.get(publicDeriverId);
+    } else {
+      const promise = this._refreshTransactionData(request);
+      this.ongoingRefreshing.set(publicDeriverId, promise);
+      await promise;
+      this.ongoingRefreshing.delete(publicDeriverId);
```

Comment:
This better to be in a `try {} finally {}` so in case of a promise throw we will also delete it from the map. if I got the idea correctly